main.tf

```diff
@@ -1,64 +1,100 @@
-# ------------------------------------------------------------------------------------------------
+# -------------------------------------------------------------------------------------------------
+# Set module requirements
+# -------------------------------------------------------------------------------------------------
+
+terraform {
+  # >= v0.12.6
+  required_version = ">= 0.12.6"
+}
+
+
+# -------------------------------------------------------------------------------------------------
+# Create defined Policies
+# -------------------------------------------------------------------------------------------------
+
+# Create customer managed policies
+resource "aws_iam_policy" "policies" {
+  for_each = local.policies
+
+  name        = lookup(each.value, "name")
+  path        = lookup(each.value, "path", "") == "" ? var.policy_path : lookup(each.value, "path")
+  description = lookup(each.value, "desc", "") == "" ? var.policy_desc : lookup(each.value, "desc")
+  policy      = templatefile(lookup(each.value, "file"), lookup(each.value, "vars"))
+}
+
+
+# -------------------------------------------------------------------------------------------------
 # Create Roles
-# ------------------------------------------------------------------------------------------------
+# -------------------------------------------------------------------------------------------------
 
+# Create roles
 resource "aws_iam_role" "roles" {
-  count = "${var.role_count}"
+  for_each = { for role in var.roles : role.name => role }
 
-  name        = "${lookup(var.roles[count.index], "name")}"
-  path        = "${lookup(var.roles[count.index], "path", "") == "" ? var.role_path : lookup(var.roles[count.index], "path")}"
-  description = "${lookup(var.roles[count.index], "desc", "") == "" ? var.role_desc : lookup(var.roles[count.index], "desc")}"
+  name        = lookup(each.value, "name")
+  path        = lookup(each.value, "path", "") == "" ? var.role_path : lookup(each.value, "path")
```

Comment:
Not sure if this works, but if you can set path to null in tfvars one can do this
```
path        = lookup(each.value, "path", var.role_path )
```

Reply:
Unfortunately not, will result in removing it:
```hcl
     - description           = "Managed by Terraform" -> null
```